planning/autoware_freespace_planner/include/autoware/freespace_planner/utils.hpp

```diff
@@ -58,13 +58,15 @@ size_t get_next_target_index(
   const size_t current_target_index);
 
 Trajectory get_partial_trajectory(
-  const Trajectory & trajectory, const size_t start_index, const size_t end_index);
+  const Trajectory & trajectory, const size_t start_index, const size_t end_index,
+  const rclcpp::Clock::SharedPtr clock);
```

Comment:
you need to update the [unit test](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/autoware_freespace_planner/test/test_freespace_planner_utils.cpp#L215) as well.

I would suggest setting the timestamp on the calling side after returning the trajectory from this function.
Then you don't need to pass the clock ptr and you don't need to worry about modifying the unit tests.

Reply:
Hi @mkquda, I've updated the missed unit tests to provide a shared clock ptr. The only reason for keeping it this way is so that if we want to develop other modules that make use of the freespace planner then they can treat it like a black box and not have to worry about doing sanity checks with the output.